frontend/packages/knative-plugin/src/components/add/EventSourcePage.tsx

```diff
@@ -2,34 +2,36 @@ import * as React from 'react';
 import { Helmet } from 'react-helmet';
 import { RouteComponentProps } from 'react-router';
 import { PageBody } from '@console/shared';
-import { PageHeading, Firehose } from '@console/internal/components/utils';
-import { ProjectModel } from '@console/internal/models';
+import { PageHeading } from '@console/internal/components/utils';
 import NamespacedPage, {
   NamespacedPageVariants,
 } from '@console/dev-console/src/components/NamespacedPage';
 import { QUERY_PROPERTIES } from '@console/dev-console/src/const';
 import EventSource from './EventSource';
+import NoKnativeServiceAlert from './NoKnativeServiceAlert';
+import { knativeServingResourcesServices } from '../../utils/get-knative-resources';
 
 type EventSourcePageProps = RouteComponentProps<{ ns?: string }>;
 
 const EventSourcePage: React.FC<EventSourcePageProps> = ({ match, location }) => {
   const namespace = match.params.ns;
   const searchParams = new URLSearchParams(location.search);
-  const resources = [{ kind: ProjectModel.kind, prop: ProjectModel.id, isList: true }];
+  const knServiceResource = { ...knativeServingResourcesServices(namespace)[0], limit: 1 };
```

Comment:
I think you should still move this into `NoKnativeServiceAlert` and pass the namespace as a prop.
There's no need for the parent to have to deal with passing along a firehose query because then there's no constrain on the resource or limit.
If you just pass the namespace, internal to `NoKnativeServiceAlert` it knows exactly what to look for.